jakarta-eclipse/src/org/jakarta/jdt/BeanValidationDiagnosticsCollector.java

```diff
@@ -0,0 +1,326 @@
+					!fieldType.equals(Signature.SIG_SHORT) &&
+					!fieldType.equals(Signature.SIG_INT) &&
+					!fieldType.equals(Signature.SIG_LONG)) {
+				
+				diagnostics.add(completeDiagnostic(new Diagnostic(fieldAnnotationrange,
+						"The @" + annotation.getElementName() + " annotation can only be used on \n- BigDecimal \n- BigInteger"
+						+ "\n- byte, short, int, long (and their respective wrappers) \n type fields.")));	
+			}
+		} else if (annotation.getElementName().equals(NEGATIVE)|| annotation.getElementName().equals(NEGATIVE_OR_ZERO) || 
+				annotation.getElementName().equals(POSITIVE)|| annotation.getElementName().equals(POSTIVE_OR_ZERO)) {
+
+			if (	!fieldType.equals(getSignatureFormatOfType(BIG_DECIMAL)) &&
+					!fieldType.equals(getSignatureFormatOfType(BIG_INTEGER)) &&
+					!fieldType.equals(getSignatureFormatOfType(BYTE)) &&
+					!fieldType.equals(getSignatureFormatOfType(SHORT)) &&
+					!fieldType.equals(getSignatureFormatOfType(INTEGER)) &&
+					!fieldType.equals(getSignatureFormatOfType(LONG)) &&
+					!fieldType.equals(getSignatureFormatOfType(FLOAT)) &&
+					!fieldType.equals(getSignatureFormatOfType(DOUBLE)) &&
+					!fieldType.equals(Signature.SIG_BYTE) &&
+					!fieldType.equals(Signature.SIG_SHORT) &&
+					!fieldType.equals(Signature.SIG_INT) &&
+					!fieldType.equals(Signature.SIG_LONG) &&
+					!fieldType.equals(Signature.SIG_FLOAT) &&
+					!fieldType.equals(Signature.SIG_DOUBLE)) {
+				
+				diagnostics.add(completeDiagnostic(new Diagnostic(fieldAnnotationrange,
+						"The @" + annotation.getElementName() + " annotation can only be used on \n- BigDecimal \n- BigInteger"
+						+ "\n- byte, short, int, long, float, double (and their respective wrappers) \n type fields.")));	
+			}
+		} else if (annotation.getElementName().equals(NOT_BLANK)) {
+
+			if (	!fieldType.equals(getSignatureFormatOfType(STRING)) &&
+					!fieldType.equals(getSignatureFormatOfType(CHAR_SEQUENCE))) {
+				
+				diagnostics.add(completeDiagnostic(new Diagnostic(fieldAnnotationrange,
+						"The @" + annotation.getElementName() + " annotation can only be used on String and CharSequence type fields.")));	
+			}
+		} else if (annotation.getElementName().equals(PATTERN)) {
+			
+			if (	!fieldType.equals(getSignatureFormatOfType(STRING)) &&
+					!fieldType.equals(getSignatureFormatOfType(CHAR_SEQUENCE))) {
+				
+				diagnostics.add(completeDiagnostic(new Diagnostic(fieldAnnotationrange,
+						"The @" + annotation.getElementName() + " annotation can only be used on String and CharSequence type fields.")));	
+			}
+		}
+		
+		// These ones contains check on all collection types which requires resolving the String of the type somehow
+		// This will also require us to check if the field type was a custom collection subtype which means we 
+		// have to resolve it and get the super interfaces and check to see if Collection, Map or Array was implemented 
+		// for that custom type (which could as well be a user made subtype) 
```

Comment:
if we are not currently adding diagnostics for NOT_EMPTY or SIZE, let's open a new issue for tracking purposes

Reply:
I added #63 issue.